src/appleseed/foundation/meta/tests/test_microfacet.cpp

```diff
@@ -419,5 +418,71 @@ TEST_SUITE(Foundation_Math_Microfacet)
         EXPECT_WEAK_WHITE_FURNACE_PASS(result)
     }
 
+    // Expect StdMDF and GGXMDF to be the same for gamma = 2
+
+    TEST_CASE(StdMDF_2_GGXMDF_comparison)
+    {
+        const StdMDF std;
+        const GGXMDF ggx;
+
+        size_t num_sample = 128;
+        for (size_t i = 0; i < num_sample; ++i)
+        {
+            static const size_t Bases[] = { 2 };
+            const Vector2f s = hammersley_sequence<float, 2>(Bases, num_sample, i);
+            const Vector3f v = sample_hemisphere_uniform(s);
+            const float std_G1 = std.G1(v, Vector3f(0.0f, 1.0, 0.0f), 0.5, 0.5, 2);
+            const float ggx_G1 = ggx.G1(v, Vector3f(0.0f, 1.0, 0.0f), 0.5, 0.5, 2);
+            EXPECT_FEQ_EPS(std_G1, ggx_G1, 0.05f);
+        }
+    }
+
+    // Check for lambda overflow
+
+    TEST_CASE(StdMDF_lambda_overflow)
+    {
+        const StdMDF std;
+
+        float gamma_max = 40.0f;
+        float gamma_step = 2.0f;
+        size_t num_sample = 128;
+        for (float i = 10.0f; i < gamma_max; i += gamma_step)
+        {
+            for (size_t j = 0; j < num_sample; ++j)            
+            {         
+                static const size_t Bases[] = { 2 };
+                const Vector2f s = hammersley_sequence<float, 2>(Bases, num_sample, i);
+                const Vector3f v = sample_hemisphere_uniform(s);
+                const float std_G1 = std.G1(v, Vector3f(0.0f, 1.0, 0.0f), 0.5, 0.5, i);
+                EXPECT_TRUE(std::isfinite(std_G1)); // check that G1 doesn't produce NaN
```

Comment:
`std::isfinite()` is available starting with C++11. For now, you'll need to use `foundation::FP<float>::is_finite()`.